### ISSUE TITLE
150+ users conference doesn't works from mobile

### ISSUE THREAD

**Issue by iospro** [jitsi/jitsi-meet#10007]
We have the problem with mobile iOS SDK in case of vast amount of participants.
Please see the log attached and the screen screenshot describing cpu/memory usage.

In the last conference we had about 150-200 users. Every mobile connection lasts about 30 seconds, at last we see user tiles,
but after about a second some internal js errors occur and the connection stops.

<img width="299" alt="Screenshot 2021-09-23 at 16 36 43" src="https://user-images.githubusercontent.com/1611869/134518640-82469114-9dcd-4276-ad9f-1f5af0f87260.png">

[jitsi log.txt](https://github.com/jitsi/jitsi-meet/files/7218109/jitsi.log.txt)

**Comment by saghul** [comment]
What SDK version are you using?

**Comment by iospro** [comment]
iOS sdk jitsi 3.9.0

we have the same problem with android, but at first we want to get approved solution on iOS.

Also we've got problems on iOS 15 with change of audio channels, and sometimes enabled microphone button needs two repeated taps to restore sound propagation.

**Comment by saghul** [comment]
Please update to SDK 3.10.1 on iOS (or 3.10 on Android). We made significant performance improvements for supporting 500 participant conferences on mobile.

**Comment by iospro** [comment]
I've made pull in master branch, it has 3.10.0 version.

This version still give me black screen at start with the following error

**Comment by saghul** [comment]
https://github.com/jitsi/jitsi-meet/releases/tag/ios-sdk-3.10.1

This is the tag.

I cannot reproduce the build issue you mention. Have you made any changes to Jitsi Meet? Are you using Node 12 and npm 6?

**Comment by iospro** [comment]
The same problem.
Unhandled JS Exception: Can't find variable: window
for
npm@7.6.3
Node.js v15.12.0

Should I downgrade all of this to Node 12 and npm 6 with guarantee that it will work?

12 and 6 seem too old

**Comment by saghul** [comment]
You can use our prebuilt SDKs instead. I suspect something doesn't get transspiled properly. We will be updating to Node 14 / npm 7 soon, but if you need a build today, you either get a prebuilt one or downgrade.

**Comment by iospro** [comment]
Could you please give links to iOS and android pre builds to avoid misleading?

**Comment by saghul** [comment]
Android instructions here: https://jitsi.github.io/handbook/docs/dev-guide/dev-guide-android-sdk#use-pre-build-sdk-artifactsbinaries for iOS it's in CocoaPods.

**Comment by iospro** [comment]
I've added the jitsi pod

pod 'JitsiMeetSDK', '3.10.1'

Now I have downgraded Node.js
Welcome to Node.js v12.22.6.

And after jitsi start for new build I get permanently
Unhandled JS Exception: Can't find variable: window 
and black screen

**Comment by saghul** [comment]
Is your app a native app? You don't need node if you are using the pod directly in a native app.

**Comment by iospro** [comment]
Ok, I managed to start app with 3.10.1 jitsi pod on my device, no errors, thanks.

But I have the following problem with a simulator

<img width="1123" alt="Screenshot 2021-09-24 at 20 09 41" src="https://user-images.githubusercontent.com/1611869/134725821-2f65c426-a5cd-4a42-8ae3-fb4191acf8a2.png">
There is no such problem when I build project with locally compiled jitsi lib (without pod)

**Comment by saghul** [comment]
Are you on an M1 Mac? We currently don't support its simulator.

You should be able to build for the device just fine.

I hope you are not going to test 150 participants on the simulator ;-)

**Comment by iospro** [comment]
Of course :) Thank you.

But I wonder that the lib admits building under M1 Mac properly for a simulator, but in cocoa pods we have no this ability.

Hope in the future it will be done )

**Comment by saghul** [comment]
Yeah, I hope to ship the required slice for M1 simulators with the next WebRTC release.
